### PR TITLE
[PETSc] 3.8.16 - bugfix

### PR DESCRIPTION
**Pull request by boriskaus**
The previous compilation of PETSc 3.18.6 accidentally shipped the source code of the 2 included binary examples (`ex4` and `ex42`) rather than the compiled binaries.
This fixes that.
We also specifically indicate version numbers of a few additional packages (w/out this it would no longer compile).

### CODE REVIEW

File: P/PETSc/build_tarballs.jl [hunk 1]
```diff
@@ -11,6 +11,11 @@ SUITESPARSE_COMPAT_VERSION = "5.10.1"
 SUPERLUDIST_COMPAT_VERSION = "8.1.2"   
 MPItrampoline_compat_version="5.2.1"    
 
+SCALAPACK32_COMPAT_VERSION="2.2.1"
```
Comment (eschnett) [inline P/PETSc/build_tarballs.jl:14]: These compat declarations assume that all these pacakges use semver, and that PETSc can be built with all other versions that are compatible according to semver. Is that true?

Comment (boriskaus) [reply P/PETSc/build_tarballs.jl:14]: In this particular case it no longer compiles if I don’t add this as superlu_dist is only compatible with a certain version of scotch, which was updated meanwhile. 
Drives me nuts, but I guess it is the only way to make it work.

Comment (eschnett) [reply P/PETSc/build_tarballs.jl:14]: Does this mean you need to add an `=` sign in front of the version number?

Comment (boriskaus) [reply P/PETSc/build_tarballs.jl:14]: yes I think so (unless I don't understand you comment)

**Comment by giordano** [comment]
Side note, you aren't new to these kind of PRs: #5443.  I hope you'll learn for the next time :slightly_smiling_face:

**Comment by boriskaus** [comment]
Yes you’re right - sorry!

**Pull request merged**